src/icelake/icelake_utf8_common.inl.cpp

```diff
@@ -269,8 +269,13 @@ simdutf_really_inline bool process_block_utf8_to_utf16(const char *&in, char16_t
     __mmask64 bxorleading = _kxor_mask64(b, leading);
     if (_kshiftli_mask64(m234, 1) != bxorleading) { return false; }
   }
-  in += 64 - _lzcnt_u64(_pdep_u64(0xFFFFFFFF, continuation_or_ascii));
-
+  //
+  if (tail == SIMDUTF_FULL) {
+    in += 32;
+    if(int8_t(*in) <= int8_t(0xc0)) in++;
```

Comment:
This contradicts what you added to the paper: it processes 33 bytes if the extra byte is not a 2 byte lead byte, errornously also processing (and it seems like then discarding) an extra byte if it was an ASCII byte.

Reply:
I think that this code is correct. It processes 32 bytes unless the byte at index 32 is a continuation byte, in which case we include it as well for a total of 33 bytes. If `x` is an ASCII byte, then the following is false: `int8_t(x) <= int8_t(0xc0)` under two's complement. I will add a comment to the code. If you think I am wrong, please provide a test case to demonstrate. What you have put in the paper does not match what is in the code because you make it conditional on the mask, which would be maintain an unnecessarily long data dependency chain.

Reply:
I think checking if byte 31 is a 2-byte lead byte (i.e. >= 0xc0) could be more efficient: the comparison can be an unsigned one with the result being in the carry flag.  The compiler might be able to turn this into a much simpler add-with-carry or similar operation.